### PR TITLE
Add shortcuts for finding and selecting conversations

### PR DESCRIPTION
**Pull request by phamdaniel**
Focus on search bar: Cmd or Ctrl + F

Specific to OSX:
* Next convo: Cmd + Shift + ]
* previous convo: Cmd + Shift + [

On Windows and Linux:
* Next : Ctrl + Tab
* Previous: Ctrl + Shift + Tab

Didn't figure out how to trigger the loading of more conversations when at the bottom of the list.

### CODE REVIEW

**Comment by sindresorhus** [comment]
> Next convo: Cmd + Shift + ]
previous convo: Cmd + Shift + [

Why these on OS X?

I tried the Messages.app and it shows Ctrl+Tab in the menu, but both Ctrl+Tab and Cmd+} works.

File: browser.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 'use strict';
 const ipc = require('electron').ipcRenderer;
+const listSelector = 'ul[aria-label="Conversation List"] > li';
```
Comment (sindresorhus) [inline browser.js]: This won't work with non-English languages.

File: browser.js [hunk 2]
```diff
@@ -19,6 +20,29 @@ ipc.on('log-out', () => {
...
+	if (index >= 0) {
+		document.querySelectorAll(listSelector)[index].firstChild.firstChild.click();
+	}
+});
+
+function getSelectedIndex() {
+	const selected = document.querySelector('._5l-3._1ht1._1ht2');
+	const list = selected.parentNode;
+
+	return Array.prototype.indexOf.call(list.children, selected);
```
Comment (sindresorhus) [inline browser.js]: Just use `Array.from()` to convert a node list to an array.

Comment (sindresorhus) [inline browser.js]: You could also use `[].includes()`

File: menu.js [hunk 3]
```diff
@@ -148,6 +148,40 @@ const darwinTpl = [
 				type: 'separator'
 			},
 			{
+				label: 'Select Next Conversation',
+				accelerator: (() => {
+					if (process.platform === 'darwin') {
+						return 'Cmd+Shift+}';
+					}
+					return 'Ctrl+Tab';
```
Comment (sindresorhus) [inline menu.js]: This could just be a ternary.

**Comment by phamdaniel** [comment]
Hmmmm
I put those as I thought those were the OSX defaults... (they're the only one that work in Terminal)
We can switch to Ctrl + Tab if you think that's better

**Comment by phamdaniel** [comment]
Actually it's more universal to have it be Ctrl Tab as the [ and ] can be in some really weird positions on non English keyboards.

File: browser.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 'use strict';
 const ipc = require('electron').ipcRenderer;
+const listSelector = 'div[aria-label="Conversations"] > ul > li';
```
Comment (sindresorhus) [inline browser.js]: Still won't work on non-English Messenger.

**Comment by sindresorhus** [comment]
Would be useful if it would go back to 1, if you press Ctrl+Tab on the last item.

**Comment by phamdaniel** [comment]
This time it should do it.
Didn't notice that the translations were identical for the test language last time...

**Comment by sindresorhus** [comment]
Excellent. Thanks for the PR and quick turnaround :)

**Pull request closed without merging**